cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -366,7 +352,7 @@ pub trait ABIMachineSpec {
     ) -> Self::I;
 
     /// Generate a return instruction.
-    fn gen_ret() -> Self::I;
+    fn gen_ret(rets: Vec<Reg>) -> Self::I;
```

Comment:
ditto

Reply:
Likewise, the instruction retains the register list, and we build it in a `Vec` in at least one callsite, so we want to avoid the re-allocation/copy here.